src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -218,15 +227,30 @@ CATCH_RETURN()
         // Allocate enough room to have one breakpoint per code unit.
         _breakpoints.resize(_text.size());
 
+        RETURN_IF_FAILED(_AnalyzeTextComplexity());
+
         if (!_isEntireTextSimple)
         {
-            // Call each of the analyzers in sequence, recording their results.
             RETURN_IF_FAILED(_fontRenderData->Analyzer()->AnalyzeLineBreakpoints(this, 0, textLength, this));
-            RETURN_IF_FAILED(_fontRenderData->Analyzer()->AnalyzeBidi(this, 0, textLength, this));

```

Comment:
I don't think we need the separate boolean or route for "is entire text simple" anymore. I think (and hope) that if the whole thing is simple, it should follow the same code path as if any individual piece of it is simple. So it should just be the loop that either does complex analysis or simple. And if the loop only loops once, then the entire thing was simple.

Reply:
You’re right. Fixed.